core/src/main/java/org/infinispan/configuration/cache/DeadlockDetectionConfiguration.java

```diff
@@ -30,14 +33,20 @@ static AttributeSet attributeDefinitionSet() {
    /**
     * Time period that determines how often is lock acquisition attempted within maximum time
     * allowed to acquire a particular lock
+    *
+    * @deprecated Since 9.0, always returns {@code -1}.
     */
+   @Deprecated
    public long spinDuration() {
-      return attributes.attribute(SPIN_DURATION).get();
+      return spinDuration.get();
```

Comment:
The field should not be removed, it's public API, but it should be marked `@Deprecated`, too.